File: src/dataset/c/resistor_color_Exercism_leleah/openAI/ChatGPT4_resistor_color_v2.c
#include "resistor_color.h"

static const resistor_band_t expected[] = { 
    BLACK, BROWN, RED, ORANGE, YELLOW, 
    GREEN, BLUE, VIOLET, GREY, WHITE 
};

resistor_band_t color_code(resistor_band_t resistor) {
    return (resistor >= BLACK && resistor <= WHITE) ? expected[resistor] : BLACK;
}

const resistor_band_t* colors(void) {
    return expected;
}

File: src/dataset/Java/resistor-color_exercism-java-uzilan_robiworks/openAI/ChatGPT4_resistor-color_v2.java
import java.util.Map;
import java.util.HashMap;

class ResistorColor {
    private static final Map<String, Integer> colorMap = createColorMap();

    private static Map<String, Integer> createColorMap() {
        String[] colors = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < colors.length; i++) {
            map.put(colors[i], i);
        }
        return map;
    }

    int colorCode(String color) {
        return colorMap.getOrDefault(color.toLowerCase(), -1);
    }

    String[] colors() {
        return colorMap.keySet().toArray(new String[0]);
    }
}